products/views.py

```diff
@@ -137,4 +139,30 @@ def get(self, request, product_id):
 
         except Product.DoesNotExist:
             return JsonResponse({'MESSAGE' : 'Product matching query does not exist.'}, status = 404)
-            
+
+
+class MainSearchView(View):
+    def get(self, request):
+        search   = request.GET.get('keywords')
+        products = Product.objects.filter(name__icontains=unquote(search))
+
+        result = [{  
+                    'id'             : product.id,
+                    'name'           : product.name,
+                    'eng_name'       : product.eng_name,
+                    'img'            : [{
+                        'img_id'     : image.id,
+                        'img_url'    : image.url
+                    } for image in ProductImage.objects.filter(product_id = product.id)],
+                    'taste'          : [{
+                        'taste_id'   : flavor.taste.id,
+                        'taste_name' : flavor.taste.name
+                    } for flavor in TasteByProduct.objects.filter(product_id = product.id)],
+                    'roasting_date'  : product.roasting_date,
+                    'price'          : product.price
+                }for product in products]
+
+        if len(products) == 0:
```

Comment:
```suggestion
        if not products.exist():
```
위와 같이 작성한다면 더 가독성이 좋을 것 같습니다! 
또한 해당 if 문의 위치는 149번 line의 result 이전에 작성되어야 할 것 같습니다.